File: tests/golden/sandbox/recipes/gt4_persona_refusal.py
"""GT-4 — advisory persona refusal: plan-only, no edits.

The advisory persona policy short-circuits ``implement``, ``test``,
and ``verify`` to SUCCESS without doing work (see
``persona_policy.py``). The capture therefore exercises a different
shape from GT-1: only one halt (``create-plan``) before the engine
walks straight to the report step.

Cycle storyboard:

1. plan halt              → recipe sets ``state.plan``
2. report runs            → engine exits 0 with a plan-only delivery
                            report (no edits, no tests, no verify).
"""
from __future__ import annotations

from pathlib import Path
from typing import Any

from . import _helpers as h

META = {
    "gt_id": "GT-4",
    "ticket_relpath": "tickets/gt-4-persona-refusal.json",
    "persona": "advisory",
    "cycle_cap": 3,
}


def build_recipe(workspace: Path) -> dict[str, Any]:
    """Advisory persona only ever hits the create-plan delegation."""

    def on_create_plan(state: dict[str, Any], record) -> dict[str, Any]:
        state["plan"] = h.standard_plan(
            "Outline modulo(a, b)",
            "Signature: modulo(a: int, b: int) -> int",
            "Edge cases: raise ValueError when b == 0",
            "Test: test_modulo_handles_zero_divisor pinning the raise",
        )
        record.recipe_notes.append(
            "advisory plan recorded; no edits will follow",
        )
        return state

    return {"create-plan": on_create_plan}


__all__ = ["META", "build_recipe"]

File: tests/golden/sandbox/recipes/gt4_persona_refusal.ts
/**
 * GT-4 — advisory persona refusal: plan-only, no edits (TS twin of
 * gt4_persona_refusal.py).
 *
 * The advisory persona policy short-circuits implement / test / verify to
 * SUCCESS without doing work, so the capture exercises only one halt
 * (`create-plan`) before the engine walks straight to the report step.
 *
 * Cycle storyboard:
 *   1. create-plan halt → recipe sets state.plan
 *   2. report runs      → engine exits 0 with a plan-only delivery report
 */
import type { RecipeModule } from './index.js';
import * as h from './_helpers.js';
import type { CycleRecord, Dict, RecipeStep } from '../runner.js';

export const META: RecipeModule['META'] = {
    gt_id: 'GT-4',
    ticket_relpath: 'tickets/gt-4-persona-refusal.json',
    persona: 'advisory',
    cycle_cap: 3,
};

export function buildRecipe(_workspace: string): Record<string, RecipeStep> {
    const onCreatePlan: RecipeStep = (state: Dict, record: CycleRecord) => {
        state['plan'] = h.standard_plan(
            'Outline modulo(a, b)',
            'Signature: modulo(a: number, b: number): number',
            'Edge cases: throw when b === 0',
            'Test: a modulo test pinning the zero-divisor throw',
        );
        record.recipe_notes.push('advisory plan recorded; no edits will follow');
        return state;
    };

    return { 'create-plan': onCreatePlan };
}

const recipe: RecipeModule = { META, buildRecipe };
export default recipe;
